pkg/client-sdk/redemption/redeem.go

```diff
@@ -90,7 +91,15 @@ func (r *CovenantlessRedeemBranch) RedeemPath() ([]string, error) {
 func (r *CovenantlessRedeemBranch) ExpiresAt() (*time.Time, error) {
 	lastKnownBlocktime := int64(0)
 
-	confirmed, blocktime, _ := r.explorer.GetTxBlockTime(r.vtxo.CommitmentTxid)
+	var confirmed bool
+	blocktime := int64(math.MaxInt64)
+	for _, txid := range r.vtxo.CommitmentTxids {
+		rConfirmed, rBlocktime, _ := r.explorer.GetTxBlockTime(txid)
+		if confirmed && rBlocktime < blocktime {
```

Comment:
if one of the commitment tx is not confirmed, we should update the rBlocktime  to `expirationFromNow := time.Now().Add(time.Minute).Add(r.vtxoTreeExpiry)`